python_modules/dagster/dagster/_core/definitions/freshness_policy_sensor_definition.py

```diff
@@ -0,0 +1,376 @@
+import inspect
+import json
+from typing import TYPE_CHECKING, Callable, Dict, List, Mapping, Optional, Sequence
+
+import pendulum
+
+import dagster._check as check
+from dagster._annotations import experimental, public
+from dagster._core.definitions.asset_selection import AssetSelection
+from dagster._core.definitions.assets import AssetsDefinition
+from dagster._core.errors import DagsterInvalidDefinitionError, DagsterInvalidInvocationError
+from dagster._core.instance import DagsterInstance
+from dagster._core.instance.ref import InstanceRef
+
+from ..decorator_utils import get_function_params
+from .events import AssetKey
+from .run_request import RunRequest, SkipReason
+from .sensor_definition import (
+    AssetMaterializationFunctionReturn,
+    DefaultSensorStatus,
+    RawSensorEvaluationFunctionReturn,
+    SensorDefinition,
+    SensorEvaluationContext,
+    is_context_provided,
+)
+from .target import ExecutableDefinition
+from .utils import check_valid_name
+
+if TYPE_CHECKING:
+    from dagster._core.definitions.repository_definition import RepositoryDefinition
+
+FreshnessPolicyMaterializationFunction = Callable[
+    ["FreshnessPolicySensorEvaluationContext"],
+    AssetMaterializationFunctionReturn,
+]
+
+
+@experimental
+class FreshnessPolicySensorEvaluationContext(SensorEvaluationContext):
+    """The context object available as the argument to the evaluation function of a
+    :py:class:`dagster.FreshnessPolicySensorDefinition`.
+
+    Users should not instantiate this object directly. To construct a
+    `FreshnessPolicySensorEvaluationContext` for testing purposes, use :py:func:`dagster.
+    build_freshness_policy_sensor_context`.
+
+    Attributes:
+        asset_keys (Sequence[AssetKey]): The asset keys that the sensor is configured to monitor.
+        repository_def (RepositoryDefinition): The repository that the sensor belongs to.
+        instance_ref (Optional[InstanceRef]): The serialized instance configured to run the schedule
+        cursor (Optional[str]): The cursor, passed back from the last sensor evaluation via
+            the cursor attribute of SkipReason and RunRequest.
+        instance (Optional[DagsterInstance]): The deserialized instance can also be passed in
+            directly (primarily useful in testing contexts).
+
+    Example:
+
+    .. code-block:: python
+
+        from dagster import freshness_policy_sensor, FreshnessPolicySensorEvaluationContext
+
+        @freshness_policy_sensor(asset_keys=[AssetKey("asset_1), AssetKey("asset_2)])
+        def the_sensor(context: FreshnessPolicySensorEvaluationContext):
+            ...
+
+    """
+
+    def __init__(
+        self,
+        instance_ref: Optional[InstanceRef],
+        cursor: Optional[str],
+        repository_def: "RepositoryDefinition",
+        asset_selection: Optional[AssetSelection],
+        asset_keys: Optional[Sequence[AssetKey]],
```

Comment:
If i understand correctly, seems like if an asset key does not belong to an asset in the same repository as this sensor, these context methods won't really return anything useful (since there is no assets definition).

Thoughts on just throwing an error if the there is no assets definition associated with a provided asset key in the repo?

Reply:
good point, done